### PR TITLE
Placing NuGet tool installer code in a common location

### PR DESCRIPTION
**Pull request by emanuelquintero**

### CODE REVIEW

File: Tasks/Common/nuget-task-common/NuGetToolGetter.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        let filteredVersions: string[] = availableVersions.map(x => x.version);
+        let versionUrlMap: Map<string, string> = new Map<string, string>();
+        availableVersions.forEach((versionInfo:INuGetVersionInfo) => {
+            versionUrlMap[versionInfo.version.toLowerCase()] = versionInfo.url;
+        });
+
+        version = toolLib.evaluateVersions(filteredVersions, versionSpec);
+        if(!version)
+        {
+            toolLib.debug('No version was found which matches ' + versionSpec);
```
Comment (pspill) [inline Tasks/Common/nuget-task-common/NuGetToolGetter.ts]: we can use taskLib.debug for all logging. Minor, just looks funny. toolLib.debug just calls in to taskLib.debug.

File: Tasks/Common/nuget-task-common/NuGetToolGetter.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            taskLib.warning(taskLib.loc("Warning_NoVersionWasFoundWhichMatches", versionSpec));
```
Comment (pspill) [inline Tasks/Common/nuget-task-common/NuGetToolGetter.ts]: these three warnings should be taskLib.error

File: Tasks/Common/nuget-task-common/NuGetToolGetter.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            taskLib.warning(availableVersions.map(x => x.version).join(";"));
+            throw new Error(taskLib.loc("Error_NuGetToolInstallerFailer", "NuGet"));
+        }
+        toolLib.debug('Found the following version from the list: ' + version);
+
+        let versionDownloadUrl: string = versionUrlMap[version.toLowerCase()];
+        if(!versionDownloadUrl)
+        {
+            toolLib.debug('No download URL was found for ' + version);
+            taskLib.warning(taskLib.loc("Warning_NoUrlWasFoundWhichMatches", version));
```
Comment (pspill) [inline Tasks/Common/nuget-task-common/NuGetToolGetter.ts]: warning should be error

File: Tasks/Common/nuget-task-common/NuGetToolGetter.ts (hunk 2 above)
Comment (pspill) [inline Tasks/Common/nuget-task-common/NuGetToolGetter.ts]: no reason to double log at debug and warning (error) levels.

File: Tasks/Common/nuget-task-common/NuGetToolGetter.ts (hunk 1 above)
Comment (emanuelquintero) [inline Tasks/Common/nuget-task-common/NuGetToolGetter.ts]: Fixed

File: Tasks/Common/nuget-task-common/NuGetToolGetter.ts (hunk 2 above)
Comment (emanuelquintero) [inline Tasks/Common/nuget-task-common/NuGetToolGetter.ts]: Fixed

File: Tasks/Common/nuget-task-common/NuGetToolGetter.ts (hunk 3 above)
Comment (emanuelquintero) [inline Tasks/Common/nuget-task-common/NuGetToolGetter.ts]: Fixed

File: Tasks/Common/nuget-task-common/NuGetToolGetter.ts (hunk 2 above)
Comment (emanuelquintero) [inline Tasks/Common/nuget-task-common/NuGetToolGetter.ts]: Fixed

**Pull request merged**